FLT/DedekindDomain/FiniteAdeleRing/BaseChange.lean

```diff
@@ -169,13 +169,28 @@ noncomputable local instance (w : HeightOneSpectrum B) :
     Algebra K (adicCompletion L w) := RingHom.toAlgebra <|
   (algebraMap L (adicCompletion L w)).comp (algebraMap K L)
 
+
 variable {B L} in
 noncomputable def adicCompletion_comap_algHom (w : HeightOneSpectrum B) :
-    letI : Algebra K (adicCompletion L w) := RingHom.toAlgebra <|
-  (algebraMap L (adicCompletion L w)).comp (algebraMap K L);
-    letI : Module K (adicCompletion L w) := Algebra.toModule
-    (HeightOneSpectrum.adicCompletion K (comap A w)) →ₐ[K] (HeightOneSpectrum.adicCompletion L w) :=
-  sorry -- use `adicCompletion_comap_ringHom` and prove it's a K-algebra homomorphism
+    (HeightOneSpectrum.adicCompletion K (comap A w)) →ₐ[K] (HeightOneSpectrum.adicCompletion L w) := by
+    use adicCompletion_comap_ringHom A K w
+    intro r
+    simp only [RingHom.toMonoidHom_eq_coe, OneHom.toFun_eq_coe, MonoidHom.toOneHom_coe,
+      MonoidHom.coe_coe]
+    have : (adicCompletion_comap_ringHom A K w) (r : adicCompletion K (comap A w))  =
+    (algebraMap L (adicCompletion L w)) (algebraMap K L r) := by
+      letI : UniformSpace L := w.adicValued.toUniformSpace;
+      letI : UniformSpace K := (comap A w).adicValued.toUniformSpace;
+      rw [adicCompletion_comap_ringHom, UniformSpace.Completion.mapRingHom]
+      have h : @UniformSpace.Completion.coe' K this r  = (r : adicCompletion K (comap A w)) := rfl
+      rw [← h, UniformSpace.Completion.extensionHom_coe
+      (UniformSpace.Completion.coeRingHom.comp (algebraMap K L))
+      (UniformSpace.Completion.mapRingHom.proof_2 (algebraMap K L)
+      (adicCompletion_comap_ringHom.proof_4 A K w) :
+      Continuous ⇑(UniformSpace.Completion.coeRingHom.comp Algebra.toRingHom)) r]
+      rfl
```

Comment:
```suggestion
      rw [show (r : adicCompletion K (comap A w)) = @UniformSpace.Completion.coe' K this r from rfl]
      apply UniformSpace.Completion.extensionHom_coe
```
All this `proof_2` and `proof_4` stuff is somehow known to the elaborator (to be honest I don't quite know what's going on here, but this proof works).